include/aws/common/byte_buf.h

```diff
@@ -438,85 +473,103 @@ static inline bool aws_byte_cursor_read_be64(struct aws_byte_cursor *cur, uint64
 }
 
 /**
- * Write specified number of bytes from array to byte cursor.
+ * Write specified number of bytes from array to byte buffer.
  *
- * On success, returns true and updates the cursor pointer/length accordingly.
- * If there is insufficient space in the cursor, returns false, leaving the
- * cursor unchanged.
+ * On success, returns true and updates the buffer length accordingly.
+ * If there is insufficient space in the buffer, returns false, leaving the
+ * buffer unchanged.
  */
-static inline bool aws_byte_cursor_write(
-    struct aws_byte_cursor *AWS_RESTRICT cur,
+static inline bool aws_byte_buf_write(
+    struct aws_byte_buf *AWS_RESTRICT cur,
     const uint8_t *AWS_RESTRICT src,
     size_t len) {
-    struct aws_byte_cursor slice = aws_byte_cursor_advance_nospec(cur, len);
+
+    struct aws_byte_cursor write_cursor;
+    write_cursor.ptr = cur->buffer + cur->len;
+    write_cursor.len = cur->capacity - cur->len;
+    struct aws_byte_cursor slice = aws_byte_cursor_advance(&write_cursor, len);
 
     if (slice.ptr) {
         memcpy(slice.ptr, src, len);
+        cur->len += len;
         return true;
     }
     return false;
 }
 
 /**
- * Copies all bytes from buffer to cursor.
+ * Copies all bytes from buffer to buffer.
  *
- * On success, returns true and updates the cursor pointer/length accordingly.
- * If there is insufficient space in the cursor, returns false, leaving the
- * cursor unchanged.
+ * On success, returns true and updates the buffer /length accordingly.
+ * If there is insufficient space in the buffer, returns false, leaving the
+ * buffer unchanged.
  */
-static inline bool aws_byte_cursor_write_from_whole_buffer(
-    struct aws_byte_cursor *AWS_RESTRICT cur,
+static inline bool aws_byte_buf_write_from_whole_buffer(
+    struct aws_byte_buf *AWS_RESTRICT cur,
     const struct aws_byte_buf *AWS_RESTRICT src) {
-    return aws_byte_cursor_write(cur, src->buffer, src->len);
+    return aws_byte_buf_write(cur, src->buffer, src->len);
 }
 
 /**
- * Copies one byte to cursor.
+ * Copies all bytes from buffer to buffer.
  *
- * On success, returns true and updates the cursor pointer/length
- accordingly.<<<<<< str-split
+ * On success, returns true and updates the buffer /length accordingly.
+ * If there is insufficient space in the buffer, returns false, leaving the
+ * buffer unchanged.
+ */
+static inline bool aws_byte_buf_write_from_whole_cursor(
```

Comment:
Ergonomic suggestion: If we're not going to modify the cursor, it may be better to pass it by value instead of as a pointer. Not only does this provide a clear indication that we're not going to modify the cursor, but it also makes it easier to do things like:

    aws_byte_buf_write_from_whole_cursor(&some_buf, aws_byte_cursor_from_c_str(some_c_string));

instead of:

    struct aws_byte_cursor tmp = aws_byte_cursor_from_c_str(some_c_string);
    aws_byte_buf_write_from_whole_cursor(&some_buf, &tmp);